src/Controller/AvailabilityController.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use App\Service\AvailableCarService;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+
+#[Route('/availabilities')]
+class AvailabilityController extends AbstractController
+{
+    #[Route('', methods: ['GET'])]
+    public function listAvailable(Request $request, AvailableCarService $availableCarService): JsonResponse
+    {
+        $stationId = $request->query->get('station_id');
+        $from = $request->query->get('from');
+        $to = $request->query->get('to');
+
+        if (!$this->validateRequest($stationId, $from, $to)) {
+            return new JsonResponse(['error' => 'Invalid or missing parameters'], JsonResponse::HTTP_BAD_REQUEST);
+        }
+
+        $availableCars = $availableCarService->getAvailableCars(
+            (int) $stationId,
+            new \DateTimeImmutable($from),
+            new \DateTimeImmutable($to)
+        );
+
+        return new JsonResponse($availableCars);
+    }
+
+    private function validateRequest(
+        $stationId,
+        $from,
+        $to,
+    ): bool {
+        if (empty($stationId) || empty($from) || empty($to)) {
+            return false;
+        }
+
+        try {
+            new \DateTimeImmutable($from);
+            new \DateTimeImmutable($to);
+        } catch (\Exception $e) {
+            return false;
+        }
+
+        return true;
```

Comment:
```suggestion
        if (!empty($stationId) && !empty($from) && !empty($to)) {
           try {
               new \DateTimeImmutable($from);
               new \DateTimeImmutable($to);
               
               return true;
           } catch (\Exception $e) {
               return false;
           }
        }

        return false;
```
Return **late** pattern :nerd_face: